hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -229,28 +228,28 @@ public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
         Path newFilePath = new Path(oldFilePath.getParent(), newFileName);
         FileUtil.copy(fs,oldFilePath, fs, newFilePath, false, fs.getConf());
         */
+        }
       }
-    }
 
-    // Compacting is very similar to applying updates to existing file
-    Iterator<List<WriteStatus>> result;
-    result = executionHelper.writeFileAndGetWriteStats(compactionHandler, operation, instantTime, scanner, oldDataFileOpt);
-    scanner.close();
```

Comment:
We already close the scanner here?

Reply:
> We already close the scanner here?
yes，put in try block，not close it explicitly，may more safely @danny0405

Reply:
> > We already close the scanner here?
 yes，put in try block，not close it explicitly，may more safely @danny0405

Reply:
If it's just a code refactoring, I would suggest we keep it as it is.

Reply:
> If it's just a code refactoring, I would suggest we keep it as it is.

it is a hotfix for the scanner not close in right way either only code refactoring，it would bring some problems if connections in some extreme scene；if consider some risk for changing it，can close the pr firstly @danny0405